instructor_notes/openscpca/introduce-analysis-modules.md

```diff
@@ -0,0 +1,89 @@
+# Instructor Notes: Introduction to analysis modules Activity
+
+
+## Learning goals
+
+- Understand the structure of `OpenScPCA-analysis` analysis modules
+- Learn how to explore and run modules, including environment setup
+- Gain experience with the data download script
+
+## Activity
+
+Instructors will explain and demonstrate this activity while participants follow along.
+We expect that participants will take all steps along with the instructor.
+Before this activity, participants should performed all set up to get started with an analysis on a Lightsail for Research instance.
+
+### Part 1: Tour of the modules on GitHub
+
+First, we'll orient participants to analysis modules in the browser:
+
+- Nagivate to GitHub where analysis modules are stored: <https://github.com/AlexsLemonade/OpenScPCA-analysis/tree/main/analyses> and
+- Nagivate to the docs <https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/>
+  - Show participants where they can find more information about creating and writing analysis modules, and compute/software requirements
+  - Make sure to explain that the requirements pages also explain how to manage dependencies within modules
+- Navigate to GitHub to show and explain the contents of the `hello-python` module: <https://github.com/AlexsLemonade/OpenScPCA-analysis/tree/main/analyses/hello-python>
+  - Particularly focus on how documentation in the README relates to contents of the module itself
+  - Discuss: What can we learn about the module from the README?
+
+### Part 2: Prepare to run the `hello-python` module
+
+According to the README instructions, we'll need to take two steps before running.
+Participants should follow along and type/code with us to perform these steps:
+  - Download data
+  - Prepare the conda environment
+
+#### Download data
+
+- Introduce the data download script, including running `./download-data.py --help` to explain the full script usage
+  - Note the differences between running on LSfR vs locally:
+    - Locally they will need to be signed into AWS with SSO
+    - Locally they may need the `--profile` flag
+  - Emphasize the `--dryrun` flag
+- Everyone should now run the download script three times (feel free to add more flags if there's time or interest):
+  - Download test data for an opportunity to explain the benefits of developing with test data:
+    - `./download-data.py --test-data`
+  - Do a _dry run_ of a download:
+    - `./download-data.py --format AnnData --include processed --projects SCPCP000001 --dryrun`
+  - Download processed AnnData files from a single project for use with `hello-python`:
```

Comment:
It might be worth showing that if they run it a second time it won't re-download? 

They should also be shown the `data` folder structure after download: show how the data is versioned by date but linked to `current`. And that the `test` data directory can be swapped in by rerunning the test download, or switch back by rerunning without test data... This may get a bit in the weeds, but I think it is handy.